File: app/src/main/java/com/example/plantea/presentacion/adaptadores/AdaptadorCalendario.java
package com.example.plantea.presentacion.adaptadores;

import android.graphics.Color;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.plantea.R;
import com.example.plantea.dominio.CalendarioUtilidades;
import com.example.plantea.dominio.Evento;

import java.time.LocalDate;
import java.util.ArrayList;

public class AdaptadorCalendario extends RecyclerView.Adapter<AdaptadorCalendario.ViewHolderCalendario>{

    private final ArrayList<LocalDate> diasMes;
    private AdaptadorCalendario.OnItemSelectedListener listener;

    public interface OnItemSelectedListener {
        void diaSeleccionado(LocalDate fecha);
    }

    public AdaptadorCalendario(ArrayList<LocalDate> diasMes, AdaptadorCalendario.OnItemSelectedListener listener)
    {
        this.diasMes = diasMes;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolderCalendario onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.calendario_celda, parent, false);
        ViewGroup.LayoutParams layoutParams = view.getLayoutParams();
        layoutParams.height = (int) (parent.getHeight() * 0.166666666);
        return new ViewHolderCalendario(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolderCalendario holder, int position) {
        LocalDate fecha = diasMes.get(position);
        if(fecha == null){
            holder.diaMes.setText("");
            holder.vistaPrincipal.setBackgroundColor(Color.rgb(224,224,224));
        }else{
            holder.diaMes.setText(String.valueOf(fecha.getDayOfMonth()));
            if(fecha.equals(CalendarioUtilidades.fechaSeleccionada)){
                holder.vistaPrincipal.setBackgroundColor(Color.rgb(255,238,88));
            }
        }

        //Mostrar imagen del evento en el calendario
        for (int i=0; i< Evento.listaEventos.size(); i++){
            if(Evento.listaEventos.get(i).getFecha().equals(fecha)){
                holder.imagenEvento.setImageURI(Uri.parse(Evento.listaEventos.get(i).getImagen()));
                holder.imagenEvento.setVisibility(View.VISIBLE);
            }
        }
    }

    @Override
    public int getItemCount() {
        return diasMes.size();
    }



    public class ViewHolderCalendario extends RecyclerView.ViewHolder implements View.OnClickListener {

        TextView diaMes;
        View vistaPrincipal;
        ImageView imagenEvento;

        public ViewHolderCalendario(@NonNull View itemView) {
            super(itemView);
            diaMes = itemView.findViewById(R.id.lbl_celda_dia);
            vistaPrincipal = itemView.findViewById(R.id.vistaPrincipal);
            imagenEvento = itemView.findViewById(R.id.img_evento);
            itemView.setOnClickListener(this);
        }
        @Override
        public void onClick(View view) {
            LocalDate fecha = diasMes.get(getAdapterPosition());
            if (listener != null) {
                listener.diaSeleccionado(fecha);
            }
        }
    }
}

File: app/src/main/java/com/example/plantea/presentacion/adaptadores/AdaptadorCalendario.kt
package com.example.plantea.presentacion.adaptadores


import android.R.attr.height
import android.R.attr.width
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.plantea.R
import com.example.plantea.dominio.CalendarioUtilidades
import com.example.plantea.dominio.Evento
import com.example.plantea.presentacion.adaptadores.AdaptadorCalendario.ViewHolderCalendario
import com.google.android.material.card.MaterialCardView
import java.time.LocalDate


class AdaptadorCalendario(private val diasMes: ArrayList<LocalDate?>, days: Array<String>, private val listaEventos: ArrayList<Evento>, private val listener: OnItemSelectedListener?) : RecyclerView.Adapter<ViewHolderCalendario>() {

    private val daysOfWeek = days
    private val VIEW_TYPE_DAY_OF_MONTH = 1
    private val VIEW_TYPE_DAY_OF_WEEK = 2

    interface OnItemSelectedListener {
        fun diaSeleccionado(context: Context?, fecha: LocalDate)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderCalendario {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.calendario_celda, parent, false)

        val layoutParams = view.layoutParams

        if (viewType == VIEW_TYPE_DAY_OF_MONTH) {
            view.setBackgroundResource(R.drawable.round_bg)
            //set item height to the same width
           // layoutParams.height = layoutParams.width
        }

        return ViewHolderCalendario(view)
    }

    override fun onBindViewHolder(holder: ViewHolderCalendario, position: Int) {
        if (position < daysOfWeek.size) {
            // Bind the days of the week to the corresponding positions
            holder.diaMes.text = daysOfWeek[position]
            holder.vistaPrincipal.isClickable = false

        } else {
            val fecha = diasMes[position-daysOfWeek.size]
            //holder.diaMes.setTextColor(holder.itemView.context.resources.getColor(R.color.md_theme_dark_surface))

            //Mostrar imagen del evento en el calendario
            for (i in listaEventos.indices) {
                if (listaEventos[i].fecha == fecha) {
                    holder.vistaPrincipal.setBackgroundResource(R.drawable.round_bg_evento)
                }
            }

            if (fecha == null) {
                holder.diaMes.text = ""
                holder.vistaPrincipal.setBackgroundResource(R.drawable.round_bg_none)
            } else {
                holder.diaMes.text = fecha.dayOfMonth.toString()
                if (fecha == CalendarioUtilidades.fechaSeleccionada) {
                    holder.vistaPrincipal.setBackgroundResource(R.drawable.round_bg_selected)
                }
            }
        }
    }

    override fun getItemViewType(position: Int): Int {
        return if (position < daysOfWeek.size) {
            // View type for days of the week
            VIEW_TYPE_DAY_OF_WEEK
        } else {
            // View type for days of the month
            VIEW_TYPE_DAY_OF_MONTH
        }
    }

    override fun getItemCount(): Int {
        return diasMes.size + daysOfWeek.size
    }

    inner class ViewHolderCalendario(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        var diaMes: TextView
        var vistaPrincipal: MaterialCardView

        init {
            diaMes = itemView.findViewById(R.id.lbl_celda_dia)
            vistaPrincipal = itemView.findViewById(R.id.vistaPrincipal)
            vistaPrincipal.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            val fecha = diasMes[bindingAdapterPosition-daysOfWeek.size]
            if (fecha != null) {
                listener?.diaSeleccionado(view.context, fecha)
            }
        }
    }
}